optuna/integration/mlflow.py

```diff
@@ -1,20 +1,149 @@
+            for more details.
+        metric_name:
+            Name of the metric. Since the metric itself is just a number,
+            `metric_name` can be used to give it a name. So you know later
+            if it was roc-auc or accuracy.
+            In case of a multi-objective optimization, a list of names should be provided.
+        nest_trials:
+            Flag indicating whether or not trials should be logged as
+            nested runs. This is often helpful for aggregating trials
+            to a particular study, under a given experiment.
+        tag_study_user_attrs:
+            Flag indicating whether or not to add the study's user attrs
+            to the mlflow trial as tags. Please note that when this flag is
+            set, key value pairs in study.user_attrs will supersede existing tags.
+    """
+
+    def __init__(
+        self,
+        tracking_uri: Optional[str] = None,
+        metric_name: Union[str, Sequence[str]] = "value",
+        nest_trials: bool = False,
+        tag_study_user_attrs: bool = False,
+    ) -> None:
+
+        _imports.check()
+
+        self._tracking_uri = tracking_uri
+        self._metric_name = metric_name
+        self._nest_trials = nest_trials
+        self._tag_study_user_attrs = tag_study_user_attrs
+
+    def initialize_experiment(self, study: optuna.study.Study) -> None:
+        """Initialize a MLFlow experiment with the study name.
+
+        If a tracking uri has been provided, MLFlow will be initialized to use it.
+
+        Args:
+            study (optuna.study.Study): Study to be tracked in MLFlow
+        """
+        # This sets the tracking_uri for MLflow.
+        if self._tracking_uri is not None:
+            mlflow.set_tracking_uri(self._tracking_uri)
+
+        # This sets the experiment of MLflow.
+        mlflow.set_experiment(study.study_name)
+
+    def set_tags(self, trial: optuna.trial.BaseTrial, study: optuna.study.Study) -> None:
+        """Sets the Optuna tags for the current MLFlow run
+
+        Args:
+            trial (optuna.trial.BaseTrial): Trial to be tracked
+            study (optuna.study.Study): Study to be tracked
+        """
+        tags: Dict[str, str] = {}
+        tags["number"] = str(trial.number)
+        tags["datetime_start"] = str(trial.datetime_start)
+        if isinstance(trial, optuna.trial.FrozenTrial):
+            tags["datetime_complete"] = str(trial.datetime_complete)
+
+            # Set state and convert it to str and remove the common prefix.
+            trial_state = trial.state
+            if isinstance(trial_state, TrialState) and trial_state.is_finished():
+                tags["state"] = str(trial_state).split(".")[-1]
+
+        # Set direction and convert it to str and remove the common prefix.
+        study_direction = study.direction
+        if isinstance(study_direction, StudyDirection):
+            tags["direction"] = str(study_direction).split(".")[-1]
```

Comment:
This code block is not provided in this PR, but this indicates that the MLFlowIntegrator does not support multi-objective optimization. If the study is multi-objective, the call of `study.direction` fails.

Now, it is not consistent here because the argument of `metric_name` can be used for multi-objective optimization. I think we have 2 choices.
- Supprot multi-objective optimization. We need to fix this code block and maybe other places. (IMO, it is out of scope of this PR. It's fine to leave this task as future work.)
- Keep the current behavior. We skip the work on multi-objective optimization and leave the task as future work.

@nzw0301 @lucafurrer What do you think?

Reply:
In my understanding, "multi-metric logging" (target of this PR) and "multi-objective optimization" (out of scope of this PR) are different. Is this right?

Reply:
I started to include some support for multi-objective optimization on the fly. I think currently it is not supported at all by the MLFlow Callback, but it would be nice to have. Could be a good thing to do one thing at a time and include this in a other PR.